lfs/client.go

```diff
@@ -380,10 +378,21 @@ func doHttpRequest(req *http.Request, creds Creds) (*http.Response, *WrappedErro
 	return res, wErr
 }
 
-func doApiRequestWithRedirects(req *http.Request, creds Creds, via []*http.Request) (*http.Response, *WrappedError) {
-	res, wErr := doHttpRequest(req, creds)
+func doApiRequestWithRedirects(req *http.Request, via []*http.Request, useCreds bool) (*http.Response, *WrappedError) {
+	var creds Creds
+	if useCreds {
+		c, err := getCreds(req)
```

Comment:
Could probably push this value onto `creds` itself, since we return anyway if there's an non-nil value in `err`.